service/git/library.go

```diff
@@ -124,3 +144,19 @@ func (l *Library) repositoryStorer(url *lookout.RepositoryInfo) (
 func (l *Library) repositoryPath(url *lookout.RepositoryInfo) string {
 	return fmt.Sprintf("%s/%s", url.RepoHost, url.FullName)
 }
+
+func (l *Library) recreate(url *lookout.RepositoryInfo) (*git.Repository, error) {
```

Comment:
It looks to me like `l.get` should be done inside the lock.
Otherwise you could have:
- 2 goroutines (A, B) check the repo at the same time
- both try to lock
- A gets the lock, inits, returns the repo, unlocks
- B gets the lock, removes the repo, breaks whatever was using A's return value

Or did I misunderstand the code?

Reply:
`get` doesn't recreate repo. only `.Get` does.